ctapipe/io/hdf5merger.py

```diff
@@ -0,0 +1,467 @@
+            self.required_nodes.add(key)
+            self._append_table(other, other.root[key])
+
+        key = "/dl1/monitoring/telescope/pointing"
+        if self.monitoring and self.telescope_events and key in other.root:
+            self.required_nodes.add(key)
+            self._append_table_group(other, other.root[key])
+
+        # quality query statistics
+        key = "/dl1/service/image_statistics"
+        if key in other.root:
+            self._add_statistics_table(other, other.root[key])
+
+        key = "/dl2/service/tel_event_statistics"
+        if key in other.root:
+            for node in other.root[key]._f_iter_nodes("Table"):
+                self._add_statistics_table(other, node)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
+
+    def close(self):
+        if hasattr(self, "h5file"):
+            self.h5file.close()
+
+    def _append_subarray(self, other):
+        # focal length choice doesn't matter here, set to equivalent so we don't get
+        # an error if only the effective focal length is available in the file
+        subarray = SubarrayDescription.from_hdf(
+            other, focal_length_choice=FocalLengthKind.EQUIVALENT
+        )
+
+        if self.subarray is None:
+            self.subarray = subarray
+            self.subarray.to_hdf(self.h5file)
+
+        elif self.subarray != subarray:
+            raise CannotMerge(f"Subarrays do not match for file: {other.filename}")
+
+    def _append_table_group(self, file, input_group, filter_columns=None):
+        """Add a group that has a number of child tables to outputfile"""
+
+        if not isinstance(input_group, tables.Group):
+            raise TypeError(f"node must be a `tables.Group`, got {input_group}")
+
+        node_path = input_group._v_pathname
+        self._get_or_create_group(node_path)
+
+        for table in input_group._f_iter_nodes("Table"):
+            self._append_table(file, table, filter_columns=filter_columns)
+
+    def _append_table(self, file, table, filter_columns=None):
+        """Append a single table to the output file"""
+        if not isinstance(table, tables.Table):
+            raise TypeError(f"node must be a `tables.Table`, got {table}")
+
+        table_path = table._v_pathname
+        group_path, table_name = split_h5path(table_path)
+
+        if table_path in self.h5file:
+            output_table = self.h5file.get_node(table_path)
+            input_table = table[:]
+            if filter_columns is not None:
+                input_table = recarray_drop_columns(input_table, filter_columns)
+
+            output_table.append(input_table.astype(output_table.dtype))
+
+        else:
+            self._get_or_create_group(group_path)
+
+            if filter_columns is None:
+                self._copy_node(file, table)
+            else:
+                input_table = recarray_drop_columns(table[:], filter_columns)
```

Comment:
Just to make sure: This keeps the metadata and not just the values?

Reply:
Good catch. It's a copy of before, but I think it's not copied.

Reply:
Fixed in f2b33ec2a7b9c